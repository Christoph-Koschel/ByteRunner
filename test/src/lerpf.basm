%entry main
%include "./test/src/natives.hasm"

    ; a
    ; b
    ; t
    ; --
    ; a + (b - a) * t
lerpf:
    dup 3
    dup 3
    dup 1
    minusf
    dup 3
    multf
    plusf

    ; clean up
    swap 2
    pop
    swap 2
    pop
    swap 2
    pop
    ret

main:
    push 69.0
    push 420.0
    push 0.0
    push 1.0
    push 10.0
    divf

loop:
    dup 3
    dup 3
    dup 3
    call lerpf

    int print_f64

    swap 1
    dup 1
    plusf
    swap 1

    dup 1
    push 1.0
    swap 1
    gef

    jmpif loop
    halt