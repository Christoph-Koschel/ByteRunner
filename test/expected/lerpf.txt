69.000000
104.100000
139.200000
174.300000
209.400000
244.500000
279.600000
314.700000
349.800000
384.900000
420.000000
